### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-select and sd-option

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2061]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-select & sd-option component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-select & sd-option can be found here: [sd-select & sd-option Notes](https://www.notion.so/matuzo/sd-select-sd-option-1baf3f6e1bc081c8b446dd685d950d7a).

------------------------------------------

## Comment
**Problem:** Tags: Removing options can be confusing. Mouse users can remove tags by clicking the icon inside the button. Keyboard users can only open the list. That's not perfect, but okay. When you use a screen reader, it gets confusing because the button's label is "Remove option x." However, the button doesn't remove the option; it opens the list. That's what happens in VoiceOver or when you use the Tab key to navigate with other screen readers. When you use the virtual cursor in NVDA, you get "button, option x", and when you move on, you get "button, remove, image, remove". You can activate both, and they'll do what you expect, Again, only when you use the virtual cursor. With the Tab key, you get "remove option x". In JAWS you get “option x remove button” either way.
**Solution:** Split the button and the icon and put the icon in its own button that you position above the other button.
Example: https://designsystem.morningstar.com/components/combo-box/?version=6.0.0&tab=variations&fwk=Vue%202#multiselect
-> Enter and Space when focussing icon button should delete the option.

**Problem:** There is no feedback for screen readers when you remove a tag.
**Solution:** Use a live region to confirm the deletion.

**Problem:** listbox/combobox roles are not supported in Talkback on Android. On top of that, the screen reader announces the combobox as “disabled” because of the readonly attribute.
**Solution 1:** Feature detection (if possible) with a fallback to a native select?
**Solution 2:** Use native selects.

## New Notes
**Problem:** Building custom selects is rarely a good idea. You took care of many things, but one of the things that makes selects so complex is the variety of shortcuts and how they differ across browsers and screen readers.
**Solution:** Consider switching to [[customizable selects](https://developer.chrome.com/blog/rfc-customizable-select)](https://developer.chrome.com/blog/rfc-customizable-select) once they are ready.

## Notes
- Disabled selects/options: See [[sd-button](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21)](https://www.notion.so/sd-button-1baf3f6e1bc08198a2d6ddf4df8b1e21?pvs=21) . Everything I wrote there about disabling form elements applies here as well.

## Accessibility information
- *No Interactive Elements: Avoid including headings or interactive elements like links, buttons, or checkboxes within dropdown options.*
**I’d use a stronger phrasing here: Don’t include headings or interactive elements like links, buttons, or checkboxes within dropdown options.**
- *If multiple options can be selected, clearly announce this capability to screenreader users and offer a way to view all chosen items.*
**It’d be great if all users got that information.**
- *Be aware that group labels in select components will be neglected by most assistive devices.*
    
    **Some not most.**
    
- Avoid disabled select fields. If needed, ensure that they remain in the regular tab order but cannot be activated, allowing screen readers to announce their state and purpose. Use the native disabled attribute or appropriate ARIA attributes (e.g., aria-disabled="true").
    
    **First you say “Avoid disabled select fields” and then you say “Use the native disabled attribute”, which in contradictory. It’s probably smarter to refer to the new “visually-disabled” attribute.**

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

**Comment by coraliefeil** [comment]
Added to accessibility information in sd-select, sd-option, sd-opt-group, sd-combobox

- Don’t include headings or interactive elements like links, buttons, or checkboxes within dropdown options.
